rules/ruleset.go

```diff
@@ -1082,6 +1082,59 @@ func (r RuleSetUpdateHelper) NewUpdateMetadata(updateTime int64, updatedBy strin
 	return UpdateMetadata{updatedAtNanos: updateTime, cutoverNanos: cutoverNanos, updatedBy: updatedBy}
 }
 
+// CurrentRuleSet represents the current state of a rule set. A rule with all the latest snapshots
+// of its ruleset.
+type CurrentRuleSet struct {
+	Namespace    string
+	Version      int
+	CutoverNanos int64
+	MappingRules []*MappingRuleView
+	RollupRules  []*RollupRuleView
+}
+
+// NewCurrentRuleSet generates the current view of the given ruleset.
+func NewCurrentRuleSet(r RuleSet) (*CurrentRuleSet, error) {
```

Comment:
Can we unexport this ctor and instead make it a method (e.g., `Snapshot`) of the `RuleSet` interface?